x-pack/filebeat/input/http_endpoint: make input GA (#39410)
elastic/beats: x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc

| 401                 | Unauthorized            | Returned when basic auth, secret header, or HMAC validation fails.
| 405                 | Method Not Allowed      | Returned if methods other than POST are used.
| 406                 | Not Acceptable          | Returned if the POST request does not contain a body.
| 415                 | Unsupported Media Type  | Returned if the Content-Type is not application/json. Or if Content-Encoding is present and is not gzip.
| 500                 | Internal Server Error   | Returned if an I/O error occurs reading the request.
| 504                 | Gateway Timeout         | Returned if a request publication cannot be ACKed within the required timeout.
|=========================================================================================================================================================

The endpoint will enforce end-to-end ACK when a URL query parameter
`wait_for_completion_timeout` with a duration is provided. For example
`http://localhost:8080/?wait_for_completion_timeout=1m` will wait up

> to 1min for the event to be published to the cluster and then return

Reviewer: ```suggestion
to 1 minute for the event to be published to the cluster and then return
```

the user-defined response message. In the case that the publication

> does not happen within the timeout duration, the HTTP response will

Reviewer: ```suggestion
does not complete within the timeout duration, the HTTP response will
```

  include_headers: ["TestHeader"]
----

Common Expression Language example:
["source","yaml",subs="attributes"]
----
{beatname_lc}.inputs:
- type: http_endpoint
  enabled: true

>   listen_address: 192.168.1.1

Reviewer 2: [Suggestion]
Use the loopback IP address so people can copy/paste the config as is for testing/experimenting.
```suggestion
  listen_address: 127.0.0.1
```

Author: I think in this case documentational consistency and clarity trumps copy/paste convenience; the other examples all use 192.168.1.1 and that address is an address that is not special. Changing either of those things I think harms clarity.
